tox.ini

```diff
@@ -1,31 +1,64 @@
 [tox]
 envlist = 
-    py27,pypy,py33,py34,pypy3,cover,docs
+    py27,pypy,py34,py35,py36,py37,pypy3,
+    docs,
+    {py2,py3}-cover,coverage
 
 [testenv]
-commands = 
-    python setup.py -q dev
-    python setup.py -q test -q
-
-[testenv:cover]
+# Most of these are defaults but if you specify any you can't fall back
+# to defaults for others.
 basepython =
-    python2.7
-commands = 
-    python setup.py -q dev
-    python setup.py -q nosetests --with-xunit --with-xcoverage
-deps =
-    nosexcover
+    py27: python2.7
+    py34: python3.4
+    py35: python3.5
+    py36: python3.6
+    py37: python3.7
+    pypy: pypy
+    py2: python2.7
+    py3: python3.5
+commands =
+    pip install pyramid_chameleon[testing]
+    nosetests --with-xunit --xunit-file=nosetests-{envname}.xml {posargs:}
+
+[py-cover]
+commands =
+    pip install pyramid_chameleon[testing]
+    coverage run --source=pyramid_chameleon --parallel-mode {envbindir}/nosetests
+    coverage combine
+    coverage xml -i -o {envname}.xml
+
+[testenv:py2-cover]
+commands =
+    {[py-cover]commands}
+setenv =
+    COVERAGE_FILE=.coverage.py2
+
+[testenv:py3-cover]
+commands =
+    {[py-cover]commands}
+setenv =
+    COVERAGE_FILE=.coverage.py3
 
 # we separate coverage into its own testenv because a) "last run wins" wrt
 # cobertura jenkins reporting and b) pypy and jython can't handle any
 # combination of versions of coverage and nosexcover that i can find.
 
+[testenv:coverage]
+basepython = python3.5
+commands =
+    coverage erase
+    coverage combine
+    coverage xml -i
+    coverage report --show-missing --fail-under=100
```

Comment:
Remove `--fail-under=100` until we get a PR that fixes things and gets tests back to 100% coverage.

Reply:
Fixed in 9ff06f2